### PR TITLE
array: add S.prepend

### LINKED ISSUE #197: Add S.prepend function
Create a typesafe version of [R.prepend](http://ramdajs.com/docs/#prepend).

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #197

### CODE REVIEW

**Comment by davidchambers** [comment]
How does this look, @gilligan?

**Comment by gilligan** [comment]
:+1: looks good to me :)

File: test/prepend.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+describe('prepend', function() {
+
+  it('is a binary function', function() {
+    eq(typeof S.prepend, 'function');
+    eq(S.prepend.length, 2);
+  });
+
+  it('type checks its arguments', function() {
+    throws(function() { S.prepend('a', 'bc'); },
```
Comment (svozza) [inline test/prepend.js]: Could we test the type checking of the final argument by adding a test that uses the placeholder as the first arg?

Comment (davidchambers) [inline test/prepend.js]: What do you have in mind, Stefano? We test these failures:

```
prepend :: a -> Array a -> Array a
                ^^^^^^^
```

```
prepend :: a -> Array a -> Array a
           ^          ^
```

We don't test these failures:

```
prepend :: a -> Array a -> Array a
                           ^^^^^^^
```

```
prepend :: a -> Array a -> Array a
           ^                     ^
```

Assuming our function is implemented correctly, there's no way to test invalid return values. ;)

Comment (svozza) [inline test/prepend.js]: Don't mind me, I'm losing my mind, I completely misread the first test there.

**Pull request merged**